poh/benches/poh.rs

```diff
@@ -65,3 +77,77 @@ fn bench_poh_lock_time_per_batch(bencher: &mut Bencher) {
         poh.hash(DEFAULT_HASHES_PER_BATCH);
     })
 }
+
+#[bench]
+fn bench_poh_recorder_record_transaction_index(bencher: &mut Bencher) {
+    let ledger_path = get_tmp_ledger_path_auto_delete!();
+    let blockstore =
+        Blockstore::open(ledger_path.path()).expect("Expected to be able to open database ledger");
+    let GenesisConfigInfo { genesis_config, .. } = create_genesis_config(2);
+    let bank = Arc::new(Bank::new_for_tests(&genesis_config));
+    let prev_hash = bank.last_blockhash();
+
+    let (mut poh_recorder, _entry_receiver, _record_receiver) = PohRecorder::new(
+        0,
+        prev_hash,
+        bank.clone(),
+        Some((4, 4)),
+        bank.ticks_per_slot(),
+        Arc::new(blockstore),
+        &std::sync::Arc::new(LeaderScheduleCache::new_from_bank(&bank)),
+        &PohConfig::default(),
+        Arc::new(AtomicBool::default()),
+    );
+    let h1 = hash(b"hello Agave, hello Anza!");
+
+    poh_recorder.set_bank_with_transaction_index_for_test(bank.clone());
+    poh_recorder.tick();
+    let txs: [SanitizedTransaction; 7] = [
+        SanitizedTransaction::from_transaction_for_tests(test_tx()),
+        SanitizedTransaction::from_transaction_for_tests(test_tx()),
+        SanitizedTransaction::from_transaction_for_tests(test_tx()),
+        SanitizedTransaction::from_transaction_for_tests(test_tx()),
+        SanitizedTransaction::from_transaction_for_tests(test_tx()),
+        SanitizedTransaction::from_transaction_for_tests(test_tx()),
+        SanitizedTransaction::from_transaction_for_tests(test_tx()),
+    ];
+
+    bencher.iter(|| {
+        let _record_result = poh_recorder
+            .record(
+                bank.slot(),
+                h1,
+                txs.iter().map(|tx| tx.to_versioned_transaction()).collect(),
```

Comment:
Think the final suggestion I would have here is to use `black_box(h1),` and `black_box(txs).iter().map(|tx| tx.to_versioned_transaction()).collect),`

will just make sure the compiler isn't doing some tricks and optimizing for the fact that those are always the same.